### PR TITLE
Add Home Assistant device tracker

### PR DESCRIPTION
**Pull request by mueslo**
Maintainer: me 
Compile tested: LEDE Reboot 17.01.2 r3435-65eec8bd5f
Run tested: LEDE Reboot 17.01.2 r3435-65eec8bd5f, tested locally

Description:
New package that notifies Home Assistant on hostapd events for presence detection. See also: https://github.com/mueslo/openwrt_hass_devicetracker

### CODE REVIEW

File: net/hass/files/hass.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+config hass 'global'
+        option host '1.2.3.4:8123'
```
Comment (karlp) [inline net/hass/files/hass.conf]: if you _must_ do this, use something.example.org instead of 1.2.3.4

File: net/hass/files/hassd.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+# todo: hostapd white (black) list so we don't have to listen to all APs
+
+source "/lib/functions.sh"
+config_load hass
+
+logger -t $0 -p info "Starting up"
+
+source /usr/lib/hass/functions.sh
```
Comment (karlp) [inline net/hass/files/hassd.sh]: you've "" quoted above, but not here.  I don't care which you use, but consistency is nice.

File: net/hass/files/hass.conf (hunk 1 above)
Comment (mueslo) [reply net/hass/files/hass.conf]: I think a local-only IP would be better than a DNS name that could be registered at some point, but yeah, 1.2.3.4 is a bad idea.

File: net/hass/files/hassd.sh (hunk 2 above)
Comment (mueslo) [reply net/hass/files/hassd.sh]: you're right, just copied the upper part everywhere.

File: net/hass/files/hass.conf (hunk 1 above)
Comment (karlp) [reply net/hass/files/hass.conf]: No, I literally meant "something.example.org"  example.{com,org,net} are permanently reserved: https://tools.ietf.org/html/rfc2606  no-one can register them :)

**Comment by dibdot** [comment]
A complete non-technical feedback regarding your chosen package name "hass", an ugly german word which means "hate"/"hatred". In times of Trump, alternative facts and hatred of foreigners not a good choice ... just my very personal reflection as I first saw your PR.

**Comment by karlp** [comment]
@dibdot it's not this packager's choice, it's home assistant itself.  => hass.io  "Home ASSistant"  I'd say using the "hass" name is most consistent and least surprising for the user.

**Comment by dibdot** [comment]
@karlp thanks for clarification, still I find this name quite awful (as a german native speaker).

**Comment by mueslo** [comment]
I'm German myself, so the association is certainly not lost on me. However,
'hass' and 'HA' are the usual abbreviations, but only 'hass' is unambiguous
about the actual software package referenced.

Am 04.03.2018 3:50 nachm. schrieb "Dirk Brenken" <notifications@github.com>:

File: net/hass/files/hass.conf (hunk 1 above)
Comment (mueslo) [reply net/hass/files/hass.conf]: TIL, thanks!

Comment (mueslo) [reply net/hass/files/hass.conf]: The TLD `example.` is also reserved, so I used that.

**Comment by nstrelow** [comment]
Would like to see this integrated 😁

File: net/hass/Makefile [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+PKG_RELEASE:=1
+PKG_LICENSE:=GPL-3.0+
+PKG_MAINTAINER:=Johannes Falke <johannesfalke@gmail.com>
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/hass
+	SECTION:=net
+	CATEGORY:=Network
+	TITLE:=Wireless device tracker for Home Assistant
```
Comment (karlp) [inline net/hass/Makefile]: ok, I missed this earlier.  If this package is _just_ the wireless device tracker, and not Home Assistant itself, then Shouldn't the package name be something like "hass-device-tracker" or something?  not just "hass" itself?

File: net/hass/files/hass.conf [hunk 4]
```diff
@@ -0,0 +1,5 @@
+config hass 'global'
+        option host 'ip.or.name.example:8123'
+        option pw ''
```
Comment (karlp) [inline net/hass/files/hass.conf]: if that's password, please just use password, instead of making people guess.

File: net/hass/files/hass.init [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    for child_pid in $(pgrep -P $pid); do
+        echo "$(all_child_pids $child_pid)"
+    done
+}
+
+stop_service() {
+    for pid in $(all_child_pids $(cat $PIDFILE)); do
+        kill $pid
+    done
```
Comment (karlp) [inline net/hass/files/hass.init]: what did you do that procd isn't handling this for you? consider fixing your scripts that you don't need this sort of procedure to die politely.

File: net/hass/files/hass.init [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+reload_service()
+{
+        stop
+        start
```
Comment (karlp) [inline net/hass/files/hass.init]: 99% sure you don't need this, as you're doing the default implentation.

File: net/hass/Makefile (hunk 3 above)
Comment (nstrelow) [reply net/hass/Makefile]: my 2cts on this matter: I would agree

**Comment by nstrelow** [comment]
@karlp Thanks so much for pushing this forward 🥇

**Comment by mueslo** [comment]
Thanks for the feedback, I'll address those issues and comment back here when I think it's actually ready for merging.

File: net/hass/files/hass.init (hunk 5 above)
Comment (mueslo) [reply net/hass/files/hass.init]: The way I'm spawning processes, procd does not correctly kill all child processes, that's why I wrote this.

**Comment by nattofriends** [comment]
Any update on this?

**Comment by feckert** [comment]
@mueslo ping any progress on this?

**Comment by ohadlevy** [comment]
bump please? :)

**Comment by castillofrancodamian** [comment]
The most forgotten 😅😅😅

**Pull request closed without merging**